### ISSUE TITLE
Navtree broken for namespace-links if namespace included in groups

### ISSUE THREAD

**Issue by sstudi** [doxygen/doxygen#10568]
# Describe the bug

I have various namespaces and various links between them from pages.
All those links work properly.
When clicking on one link it open the corresponding namespace site.
But on some, the navtree is completely collapsed.
I can open it, but all subtrees are collapsed as well.
When navigating to the namespaces, the correct one is not highlighted.

After some digging I found out, that the incorrect behavior seems to affect all namespaces
where the `addtogroup` command is used as well:

```cpp
/**
 * @addtogroup MyTestGroup
 * @{
 */

/**
 * @brief Namespace inside a group
 */
namespace namespace2 {

/**
 * @brief Some class
 */
class SomeClass {
};

} // namespace namespace2

/**
 * @}
 */
```

Affected are all links to the namespaces. Also the one in the navtree itself.

# Expected behavior
For all namespaces the navtree should stay open and hightlight the correct location.

# Screenshots
Screenshots from example provided afterwards.

## navtree after clicking on link of namespace1 --> correctly opened and highlighted 
![image](https://github.com/doxygen/doxygen/assets/9448082/d168ef36-f597-44b6-ad2b-7102f4b4459e)

## navtree after clicking on link of namespace2 --> navtree closed 
One can see that there's also a link to the group `MyTestGroup` below the namespace title.
![image](https://github.com/doxygen/doxygen/assets/9448082/9dfce097-bd26-4ea0-b76f-1e0f5ee164f8)

# To Reproduce

[doxygen_namespace_navtree_example.zip](https://github.com/doxygen/doxygen/files/13975575/doxygen_namespace_navtree_example.zip)

# Version
At least doxygen 1.9.8 & doxygen 1.10.0 windows x64 and linux.

**Comment by albert-github** [comment]
In the version 1.9.3 I found the correct behavior and in version 1.9.4 this was goone.

It looks like the difference between the versions can be reduced to:
```
diff -r html_1_9_3/navtreeindex0.js html_1_9_4/navtreeindex0.js
23d22
< "namespacenamespace2.html":[0,0,0],
```

Question is why did this line disappear?, so needs some bisecting ...

**Comment by albert-github** [comment]
Bisecting gave:
```
7f514f1822de413da25bd5a6eba8938a3483e793 is the first bad commit
commit 7f514f1822de413da25bd5a6eba8938a3483e793
Date:   Sun Apr 24 13:35:58 2022 +0200

    SHOW_FILES=NO could cause broken links for grouped files

 src/index.cpp | 44 ++++++++++++++++++++------------------------
 1 file changed, 20 insertions(+), 24 deletions(-)
```


I've just pushed a proposed patch, pull request #10576

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
